### PR TITLE
feat: sort interface and class members

### PR DESCRIPTION
**Pull request by jasonchoimtt**
cc @IgorMinar

### CODE REVIEW

File: lib/serializer.ts [hunk 1]
```diff
@@ -135,9 +139,31 @@ function getSanitizedText(node: ts.Node, options: SerializationOptions): string
...
-  const children = node.getChildren();
+  let children = node.getChildren();
   if (children.length) {
-    return node.getChildren().map(n => getSanitizedText(n, options)).join('');
+    // Sort declarations under a class or an interface
+    if (node.kind === ts.SyntaxKind.SyntaxList) {
+      switch (node.parent && node.parent.kind) {
+        case ts.SyntaxKind.ClassDeclaration:
+        case ts.SyntaxKind.InterfaceDeclaration: {
+          if (children.every(node => node.kind in memberDeclarationOrder)) {
```
Comment (IgorMinar) [inline lib/serializer.ts:153]: what if a new kind is added that we don't have in our list?

File: test/unit_test.ts [hunk 2]
```diff
@@ -137,6 +137,44 @@ describe('unit test', () => {
     check({'file.d.ts': input}, expected);
   });
 
+  it('should sort interface members', () => {
```
Comment (IgorMinar) [inline test/unit_test.ts:164]: can you add a test for static members using a class?

File: test/unit_test.ts [hunk 3]
```diff
@@ -202,19 +240,19 @@ describe('unit test', () => {
...
           /**
-           * A very useful field.
-           */
-          name: string;
-          /**
            * A very interesting getter.
            */
           b: string;
+          /**
+           * A very useful field.
```
Comment (IgorMinar) [inline test/unit_test.ts:271]: :)

File: lib/serializer.ts (hunk 1 above)
Comment (jasonchoimtt) [inline lib/serializer.ts:153]: The problem is that `SyntaxList` is arbitrarily generated by the `Node#getChildren()` function, so we have to somehow make sure we are working with the right list (of member declarations). We couldn't check all the sub-classes of `Declaration` either, thanks to duck typing.

How about replacing the `every` condition with:

``` typescript
const members = node.parent.members;
if (children.every(node => members.indexOf(node) >= 0)) { ... }
```

Should be more future proof.

**Comment by IgorMinar** [comment]
lgtm

**Comment by IgorMinar** [comment]
@jasonchoimtt I published 0.1.3, can you please update angular/angular

**Pull request merged**